lightblue-client-integration-test/src/test/java/com/redhat/lightblue/client/integration/test/LockingTest.java

```diff
@@ -32,13 +36,43 @@ public void testAcquireAndRelease() throws Exception {
         assertTrue(lbLocker.release("test-lock"));
     }
 
+    @Test(expected = LightblueResponseException.class)
+    public void testPingWithError() throws Exception {
+        Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
+
+        lbLocker.ping("test-lock");
+    }
+
+    @Test(expected = LightblueResponseException.class)
+    public void testCountWithError() throws Exception {
+        Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
+
+        lbLocker.getLockCount("test-lock");
+    }
+
+    @Test(expected = LightblueResponseException.class)
+    public void testReleaseWithError() throws Exception {
+        Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
+
+        lbLocker.release("test-lock");
+    }
+
     @Test
     public void testLock() throws LightblueException, IOException {
         Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
         try (Lock lock = lbLocker.lock("test-lock")) {
             assertEquals(1, lock.getLockCount());
         }
-        assertEquals(0, lbLocker.getLockCount("test-lock"));
+
+        //assert the lock was successfully released.
+        try {
+            lbLocker.getLockCount("test-lock");
+            fail();
+        } catch (LightblueResponseException e) {
+            Error error = e.getLightblueResponse().getLightblueErrors()[0];
+            assertNotNull(error);
+            assertTrue(error.getMsg().contains("InvalidLockException: test-lock"));
+        }
```

Comment:
Lock exceptions are not exactly response exceptions. Maybe you can consider
using a different base exception, something like LockException?

On Wed, Jan 13, 2016 at 12:33 PM, Dennis Crissman <notifications@github.com>
wrote:

> In
> lightblue-client-integration-test/src/test/java/com/redhat/lightblue/client/integration/test/LockingTest.java
> <https://github.com/lightblue-platform/lightblue-client/pull/234#discussion_r49638851>
> :
>
> >      @Test
> >      public void testLock() throws LightblueException, IOException {
> >          Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
> >          try (Lock lock = lbLocker.lock("test-lock")) {
> >              assertEquals(1, lock.getLockCount());
> >          }
> > -        assertEquals(0, lbLocker.getLockCount("test-lock"));
> > +
> > +        //assert the lock was successfully released.
> > +        try {
> > +            lbLocker.getLockCount("test-lock");
> > +            fail();
> > +        } catch (LightblueResponseException e) {
> > +            Error error = e.getLightblueResponse().getLightblueErrors()[0];
> > +            assertNotNull(error);
> > +            assertTrue(error.getMsg().contains("InvalidLockException: test-lock"));
> > +        }
>
> I am not sure that a runtime exception is the way to go though. I am
> exploring the possibility of extending LightblueResponseException.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/234/files#r49638851>
> .
>